README.md

```diff
@@ -1,6 +1,162 @@
 # TypeScript Flag Upgrade
 
-A CLI tool that automatically upgrades TypeScript codebases to conform to strict compiler flags.
+A CLI tool that automatically upgrades TypeScript codebases to conform to strict compiler flags using the `ts-morph` library and TypeScript compiler API.
+
+## Getting Started
+
+### Usage
+
+To upgrade a TypeScript project, first clone this repository and navigate to its directory:
+
+```
+git clone https://github.com/googleinterns/typescript-flag-upgrade.git
+cd typescript-flag-upgrade
+```
+
+Then run the following command with the corresponding arguments:
+
+```
+npm run dev -- -p <relative-path-to-tsconfig> [-i <relative-path-to-input-dir>]
+```
+
+More specifically, the mandatory `-p` argument specifies the relative path to the `tsconfig.json` file of the TypeScript project to be upgraded. The tool will automatically retrieve all TypeScript files within the project based on the `tsconfig.json`. In the case you want to specify a separate or subdirectory to upgrade, you can specify the optional `-i` flag to override the input directory.
+
+After running the command, the tool mutatively changes the input TypeScript files to fix for any errors that it found. Before each change, the tool inserts a comment of the form: `// typescript-flag-upgrade automated fix: --[compilerFlag]`.
+
+**Important:** Before running this tool, ensure that your TypeScript project is already compiling with no errors. This means that you should set any compiler flags that are causing errors to `false`. If the input TypeScript project has compiler errors, then this tool will error out and stop execution. After running the tool, reactivate the compiler flags that were originally causing the errors, and the number of errors should have decreased.
+
+For help, run:
+
+```
+npm run dev -- --help
+```
+
+### User Flow
+
+This tool was designed to be used iteratively as a way to semi-automatically fix and speed up the upgrading process for engineers. After each run, the tool fixes as many errors as possible automatically. For errors that the tool is unable to fix automatically (see Supported Flags and Fixes), the tool logs a message to the engineer detailing the location of the error, as well as the relative priority of the error (based on how many other errors are associated with this location). After fixing the highest priority errors, the engineer can re-run this tool, which will then propogate this fix other errors associated with the high-priority fixes that the engineer just made. This process can be repeated as a way to iteratively upgrade larger TypeScript codebases.
+
+## Developer and Maintenance Guide
+
+This tool was designed, implemented, and tested within a period of 3 months, and there are many areas in which it can be improved.
+
+### Supported Flags and Fixes
```

Comment:
May be we should put a section explicitly put together "things we will not support", so people know the scope of the fixes.

usually in design doc we have "goals" and then "Non goals", so people reading the doc is more clear on what problem we are solving.